xwiki-platform-core/xwiki-platform-oldcore/src/test/java/com/xpn/xwiki/web/XWikiServletURLFactoryTest.java

```diff
@@ -544,6 +545,23 @@ public void createResourceURL()
         // Verify that the URL factory encodes each path segment.
         URL url = this.urlFactory.createResourceURL("o;ne/t?w&o/t=hr#e e", false, this.oldcore.getXWikiContext());
         assertEquals("http://127.0.0.1/xwiki/resources/o;ne/t%3Fw&o/t=hr%23e%20e", url.toString());
+
+        Map<String, Object> queryParametersMap = new LinkedHashMap<>();
+        queryParametersMap.put("cache-version", "11.1-SNAPSHOT#");
+        queryParametersMap.put("anArrayOfInt", new int[] { 42, 56 });
+        queryParametersMap.put("aListWithStringToEncode", Arrays.asList("épervier", "androïde"));
+        queryParametersMap.put("aCustomObject", new Object() {
+            @Override
+            public String toString() {
+                return "foo";
+            }
+        });
+        url = this.urlFactory.createResourceURL("o;ne/t?w&o/t=hr#e e", false, this.oldcore.getXWikiContext(),
+            queryParametersMap);
+        assertEquals("http://127.0.0.1/xwiki/resources/o;ne/t%3Fw&o/t=hr%23e%20e?cache-version=11.1-SNAPSHOT%23"
```

Comment:
@tmortagne so I added this new assert to assess the behaviour of the URLBuilder with query parameters, please ensure that's indeed what we expect.